Subdivision_method_3/include/CGAL/Subdivision_method_3/subdivision_masks_3.h

```diff
@@ -373,6 +373,99 @@ class Loop_mask_3 : public PQQ_stencil_3<PolygonMesh, VertexPointMap> {
 /// @}
 };
 
+// ======================================================================
+/*!
+\ingroup PkgSurfaceSubdivisionMethod3Ref
+
+TODO: need to find the correct name for this scheme, if it exists. And also complete the documentation.
+
+The geometry mask of Upsample subdivision. This does not change the geometry of the mesh.
+In other words, the original vertices are not moved, and the new vertices are
+the average of the incident vertices. This is useful for increasing the resolution
+of a mesh without changing its geometry.
+
+A stencil determines a source neighborhood
+whose points contribute to the position of a refined point.
+The geometry mask of a stencil specifies
+the computation on the nodes of the stencil.
+`Upsample_mask_3` implements the geometry masks of
+Upsample subdivision on a triangulated model of `MutableFaceGraph`,
+such as `Polyhedron_3` and `Surface_mesh`.
+
+\tparam PolygonMesh must be a model of the concept `MutableFaceGraph`. Additionally all faces must be triangles.
+\tparam VertexPointMap must be a model of `WritablePropertyMap` with value type `Point_3`
+
+\cgalModels{PTQMask_3}
+
+\sa `CGAL::Subdivision_method_3`
+
+*/
+template <class PolygonMesh,
+          class VertexPointMap = typename boost::property_map<PolygonMesh,
+                                                              vertex_point_t>::type >
+class Upsample_mask_3 : public PQQ_stencil_3<PolygonMesh, VertexPointMap> {
```

Comment:
This seems to be exactly Linear_mask_3, with the small detail that `vtp` in linear_mask_3::border_node() needs to also be computed. That is:

```cpp
  void border_node(halfedge_descriptor edge, Point& ept, Point& vtp){
    edge_node(edge, ept);
    vertex_node(target(edge, *(this->pmesh)), vtp); // <-------- this is new
  }
```

 I don't know why it was not, and as far as I see it was just forgotten initially (https://github.com/CGAL/cgal/blame/2cec4e33c391e2da90403e3706e9fadd6c414dec/Subdivision_method_3/include/CGAL/Subdivision_mask_3.h#L107). It doesn't impact masks who depend on Linear_mask_3 because they overload the border_edge() functions.